instance/models/openedx_instance.py

```diff
@@ -314,23 +325,35 @@ def get_load_balancer_configuration(self, triggered_by_instance=False):
         The triggered_by_instance flag indicates whether the reconfiguration was initiated by this
         instance, in which case we log additional information.
         """
-        if not self.active_appserver:
+        active_appservers = self.get_active_appservers()
+        if not active_appservers.exists():
             return self.get_preliminary_page_config(self.ref.pk)
-        ip_address = self.active_appserver.server.public_ip
-        if not ip_address:
+
+        # Create the haproxy backend configuration from the list of active appservers
+        appserver_vars = []
+        for appserver in active_appservers:
+            server_name = "appserver-{}".format(appserver.pk)
+            ip_address = appserver.server.public_ip
+            if ip_address:
+                appserver_vars.append(dict(ip_address=ip_address, name=server_name))
+            else:
+                appserver.logger.error(
+                    "Active appserver does not have a public IP address. This should not happen."
+                )
```

Comment:
We are will have to change how this case is handled to fix the problem we experienced yesterday, but this is out of scope for the current PR.